pkg/admin/data/backup.go

```diff
@@ -0,0 +1,136 @@
+package data
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+
+	"github.com/openshift/microshift/pkg/config"
+	"github.com/openshift/microshift/pkg/util"
+
+	"k8s.io/klog/v2"
+)
+
+var (
+	cpArgs = []string{
+		"--verbose",
+		"--recursive",
+		"--preserve",
+		"--reflink=auto",
+	}
+)
+
+// MakeBackup backs up MicroShift data (/var/lib/microshift) to
+// target/name/ (e.g. /var/lib/microshift-backups/backup-00001).
+func makeBackup(cfg BackupConfig) error {
+	if cfg.Storage == "" {
+		return fmt.Errorf("backup storage must not be empty")
```

Comment:
Can these validation steps be performed at a place where we could provide an error message that includes the command line option they need to use? We wouldn't want the CLI interface to leak into this low level module, but if I saw this error text I'm not sure I would know immediately what I should have done differently.

Reply:
Yes, I'll add that. But I'd like to leave these here as well (speaking from experience, I already managed to pass empty deployID when working on https://github.com/openshift/microshift/pull/1826/ resulting in backup being in `cfg.Storage` rather than `cfg.Storage/cfg.Name`)